### PR TITLE
fix: prevent handlers unsubscribed in the middle of setState from being called

### LINKED ISSUE #4: iterating over subscribers should check to see if id still exists and if it's a function
The 2.0.1 patch release is throwing in error on our site whereas 2.0.0 was fine.

The problem is occurring due to an earlier listener causing a later listener on in the listeners object to no longer exist, thus throwing an error.

Perhaps the iteration over the listeners should check to see if it is a function before calling it?

### PR DESCRIPTION
**Pull request by vesparny**
Closes #4

### CODE REVIEW

**Comment by vesparny** [comment]
@brucewpaul I managed to reproduce your issue.
The last unit test should address exactly the issue you pointed out.

Mind reviewing?

File: index.js [hunk 1]
```diff
@@ -7,14 +7,24 @@ export default function createBroadcast (initialState) {
 
   const setState = state => {
     _state = state
-    Object.keys(listeners).forEach(id => listeners[id](_state))
+    const keys = Object.keys(listeners)
+    for (let i = 0; i < keys.length; i += 1) {
+      // if a listener gets unsubscribed during setState we just skip it
+      if (typeof listeners[keys[i]] !== 'undefined') {
```
Comment (brucewpaul) [inline index.js:13]: shouldn't it specifically check to see if it is a function? If you subscribe with a non-function, it would still cause an error.

Comment (vesparny) [reply index.js:13]: I've added a check to throw an error in case you do not pass a function to subscribe

Comment (brucewpaul) [reply index.js:13]: 👍

**Comment by brucewpaul** [comment]
Looks good, thanks for the update!

**Pull request merged**